File: php/v1.1/examples/GetAllCustomChannels.php
<?php
/*
 * Copyright 2014 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * This example gets all custom channels in an ad client.
 *
 * Tags: customchannels.list
 */
class GetAllCustomChannels {
  // Ad Exchange accounts are big, so for the purpose of this sample, we limit
  // the maximum number of pages we allow it to get.
  const MAX_PAGES = 2;

  /**
   * Gets all custom channels in an ad client.
   *
   * @param $service Google_Service_AdExchangeSeller AdExchange Seller service
   *     object on which to run the requests.
   * @param $adClientId string the ID for the ad client to be used.
   * @param $maxPageSize int the maximum page size to retrieve.
   * @return array the last page of retrieved custom channels.
   */
  public static function run($service, $adClientId, $maxPageSize) {
    $separator = str_repeat('=', 80) . "\n";
    print $separator;
    printf("Listing all custom channels for ad client %s\n", $adClientId);
    print $separator;

    $optParams['maxResults'] = $maxPageSize;

    $pageToken = null;
    $customChannels = null;
    $pageNumber = 0;
    do {
      if ($pageNumber == self::MAX_PAGES) {
        break;
      }
      $optParams['pageToken'] = $pageToken;
      $result = $service->customchannels->listCustomchannels($adClientId,
          $optParams);
      if (!empty($result['items'])) {
        $customChannels = $result['items'];
        foreach ($customChannels as $customChannel) {
          printf("Custom channel with code \"%s\" and name \"%s\" was found.\n",
              $customChannel['code'], $customChannel['name']);
        }
        if (isset($result['nextPageToken'])) {
          $pageToken = $result['nextPageToken'];
        }
      } else {
        print "No custom channels found.\n";
      }
      $pageNumber++;
    } while ($pageToken);
    print "\n";

    return $customChannels;
  }
}

File: ruby/v2.0/get_all_custom_channels.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Author:: api.Dean.Lukies@gmail.com (Dean Lukies)
#
# Copyright:: Copyright 2014, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example gets all custom channels in an ad client.
#
# To get ad clients, run get_all_ad_clients.rb.
#
# Tags: customchannels.list

require 'adexchangeseller_common'
require 'optparse'

# The maximum number of results to be returned in a page.
MAX_PAGE_SIZE = 50

def get_all_custom_channels(adexchangeseller, options)
  request = adexchangeseller.accounts.customchannels.list(
      :accountId => ACCOUNT_ID,
      :adClientId => options[:ad_client_id],
      :maxResults => MAX_PAGE_SIZE)

  loop do
    result = request.execute

    result.data.items.each do |custom_channel|
      puts 'Custom channel with id "%s" and name "%s" was found.' %
        [custom_channel.id, custom_channel.name]

      if custom_channel['targetingInfo']
        puts '  Targeting info:'
        targeting_info = custom_channel.targetingInfo
        if targeting_info['adsAppearOn']
          puts '    Ads appear on: %s' % targeting_info.adsAppearOn
        end
        if targeting_info['location']
          puts '    Location: %s' % targeting_info.location
        end
        if targeting_info['description']
          puts '    Description: %s' % targeting_info.description
        end
        if targeting_info['siteLanguage']
          puts '    Site language: %s' % targeting_info.siteLanguage
        end
      end
    end

    break unless result.next_page_token
    request = result.next_page
  end
end


if __FILE__ == $0
  adexchangeseller = service_setup()

  options = {}

  optparse = OptionParser.new do |opts|
    opts.on('-c', '--ad_client_id AD_CLIENT_ID',
            'The ID of the ad client for which to generate a report') do |id|
      options[:ad_client_id] = id
    end
  end

  begin
    optparse.parse!
    unless options[:ad_client_id].nil? ^ options[:report_id].nil?
      raise OptionParser::MissingArgument
    end
  rescue OptionParser::MissingArgument
    puts 'Please specify an ad_client_id.'
    puts optparse
    exit
  end

  get_all_custom_channels(adexchangeseller, options)
end
